### ISSUE TITLE
Regression: PlotDataItem.setData([],[]) no longer works

### ISSUE THREAD

**Issue by 2xB** [pyqtgraph/pyqtgraph#1445]
### Short description
PlotDataItem.setData([],[]) no longer works since merge of #1140 (Tagging @NilsNemitz ).

### Code to reproduce

```python
import pyqtgraph as pg

app = pg.mkQApp()
w = pg.PlotWidget()
p = w.plot([1,2], [2,2])
p.setData([], [])
w.show()

app.exec_()
```

### Expected behavior
PlotDataItem should be cleared.

### Real behavior
Error is thrown:

```
Traceback (most recent call last):
  File "test_.py", line 11, in <module>
    p.setData([], [])
  File "pyqtgraph/graphicsItems/PlotDataItem.py", line 530, in setData
    self.updateItems()
  File "pyqtgraph/graphicsItems/PlotDataItem.py", line 552, in updateItems
    x,y = self.getData()
  File "pyqtgraph/graphicsItems/PlotDataItem.py", line 664, in getData
    data_range = self.dataRect()
  File "pyqtgraph/graphicsItems/PlotDataItem.py", line 689, in dataRect
    ymin = np.nanmin(self.yData)
  File "<__array_function__ internals>", line 5, in nanmin
  File "/usr/lib/python3.8/site-packages/numpy/lib/nanfunctions.py", line 319, in nanmin
    res = np.fmin.reduce(a, axis=axis, out=out, **kwargs)
ValueError: zero-size array to reduction operation fmin which has no identity
```

### Tested environment(s)

 * PyQtGraph version: Since commit 65e90faec58d90a0394e0fd0a28f7007d43ef814
 * Qt Python binding: PyQt5 5.15.1 Qt 5.15.1
 * Python version: 3.8.5
 * NumPy version: 1.19.2
 * Operating system: Linux 5.8.16-2-MANJARO x86_64 20.1.2 Mikah
 * Installation method: Executed in folder

**Comment by NilsNemitz** [comment]
Dear @2xB , I guess that is not supposed to happen.
Thank you for the page. I have prepared pull request #1446 that should fix this.

There is some strangeness to the setData code which allows the internal data to be set to some unexpected values. I think it would be best to clean that up a little, but I hope one of the developers can weigh in with some suggestions.

**Comment by 2xB** [comment]
Hello @NilsNemitz, thank you very much for this quick response, even with a PR!

Development of `pyqtgraph` is currently rather community-driven, so if you see direct potential improvements in `setData` and have the time for it, you may just feel free to suggest an overhaul e.g. in a PR or start suggestions and discussion in a new issue. Just a warning: The most common reason for overly complex code in `pyqtgraph` is a huge amount of bug-fixing patches. That - in combination with the fact that changes in the `pyqtgraph` API are aimed at keeping the amount of API breaking changes minimal - makes overhauls rather involved.

Thank you once more for your fix!

**Comment by 2xB** [comment]
Fixed by #1140 . Btw. @NilsNemitz, a trick you might be interested in: If you finish your first message in a Pull Request with "Fixes #123" or "Closes #123", issue #123 is automatically closed after the PR is merged! Example: #1443 for #1277.
